### PR TITLE
Add comprehensive backend tests for Reservation module including repository, service, controller, and integration

### PR DESCRIPTION
**Pull request by marina-ivanovic**
This PR adds extensive test coverage and improvements to the Reservation module in the backend:

* **ReservationRepositoryTest**: Unit tests for custom queries ensuring correct fetching of reservations by event ID and edge cases.
* **ReservationServiceTest**: Tests for business logic methods including creating, accepting, rejecting, and canceling reservations with validation of reservation states.
* **ReservationControllerTest**: Integration tests for REST endpoints to verify correct handling of reservation operations and error cases.
* **Test Data**: Added sample reservations and notifications to `data.sql` for consistent testing and development environment.
* **Selenium UI Tests**: Added end-to-end tests for event search and filtering functionality using Selenium WebDriver.
* Refactoring and fixing local merge conflicts for a clean and stable test suite.

### CODE REVIEW

File: eventPlanner/src/test/java/com/ftn/iss/eventPlanner/selenium/page/FilterEventsPage.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        wait.until(ExpectedConditions.visibilityOfElementLocated(optionLocator));
+
+        // Scroll and click
+        WebElement option = driver.findElement(optionLocator);
+        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", option);
+        new Actions(driver).moveToElement(option).click().perform();
+    }
+
+    public void setMinRating(double rating) {
+        WebElement sliderInput = ratingInput; // već nađen element inputa u slideru
```
Comment (milicaradicc) [inline eventPlanner/src/test/java/com/ftn/iss/eventPlanner/selenium/page/FilterEventsPage.java]: Remove or translate comments

**Review by mijatkrivokapic — APPROVED** [review]
Good job! All tests passed and they cover all edge cases.
